### PR TITLE
Small fix for Streamlet name and extending UT coverages

### PR DESCRIPTION
**Pull request by erenavsarogullari**
This PR aims the following changes:

- `Streamlet`.`setName` can be robust for **blank** values. Also UT coverage is added for invalid cases.
- `LocalFileSystemContextTest` UT coverage is added (to avoid changing hardcoded fileSystemDirectory default value.)
- Minor javadoc updates.

### CODE REVIEW

**Comment by lucperkins** [comment]
@erenavsarogullari Nice! I just noticed some of these spelling errors the other day. LGTM, but make sure to wait on approval from @srkukarni.

**Comment by erenavsarogullari** [comment]
Thanks @lucperkins and @srkukarni for the review ;)

**Comment by srkukarni** [comment]
@erenavsarogullari Can you pl resolve the merge conflict?

**Comment by kramasamy** [comment]
@srkukarni - where is the merge conflict?

File: heron/api/tests/java/com/twitter/heron/streamlet/impl/StreamletImplTest.java [hunk 1]
```diff
@@ -262,9 +263,9 @@ public void testComplexBuild() throws Exception {
 
   @Test
   public void testResourcesBuilder() {
-    Resources defaultResoures = Resources.defaultResources();
-    assertEquals(0, Float.compare(defaultResoures.getCpu(), 1.0f));
-    assertEquals(defaultResoures.getRam(), ByteAmount.fromMegabytes(100));
+    Resources defaultResources = Resources.defaultResources();
+    assertEquals(0, Float.compare(defaultResources.getCpu(), 1.0f));
+    assertEquals(defaultResources.getRam(), 104857600);
```
Comment (kramasamy) [inline heron/api/tests/java/com/twitter/heron/streamlet/impl/StreamletImplTest.java]: @erenavsarogullari - ByteAmount is a convenience class that we use to make it more readable. Any idea why we need to change ByteAmount?

**Comment by srkukarni** [comment]
I saw a merge conflict a little earlier. Also it seems I cannot see the travis pass link. Does anyone know what happened?

**Comment by srkukarni** [comment]
We want to avoid dependencies with any layer. Thats why the new resource class has setRamInMB and setRamInGB helper methods. But thats outside this pr scope

**Comment by kramasamy** [comment]
@srkukarni - this is just test where the dependencies do not mater?

**Comment by srkukarni** [comment]
But still would rather use the setInMB and GB alternatives which are as clear

**Comment by erenavsarogullari** [comment]
Hi @kramasamy, @srkukarni and @nwangtw,
Sorry, yesterday, i was offline due to timezone diff. Probably, my patch had rebase problem by amending commit message. Many thanks for the following this and sorry again for inconvenience.

**Pull request merged**